projects/control-service/projects/base/src/main/java/com/vmware/taurus/security/SecurityConfiguration.java

```diff
@@ -159,6 +169,40 @@ private SecurityFilterChain enableSecurity(
     return http.build();
   }
 
+  @Bean
+  @ConditionalOnExpression("${datajobs.security.oauth2.enabled}")
+  JwtIssuerAuthenticationManagerResolver byIssuer(JwtAuthenticationConverter converter) {
+    Map<String, AuthenticationManager> managers = new HashMap<>();
+    List<String> issuers = Lists.asList(issuer, extraIssuers);
+    for (String issuer : issuers) {
+
+      if (issuer.isBlank()) {
+        continue;
+      }
+
+      JwtDecoder decoder = jwtDecoder(issuer);
+      JwtAuthenticationProvider provider = new JwtAuthenticationProvider(decoder);
+      provider.setJwtAuthenticationConverter(converter);
+      managers.put(issuer, provider::authenticate);
+    }
+    return new JwtIssuerAuthenticationManagerResolver(managers::get);
+  }
+
+  @Bean
+  @ConditionalOnExpression(
+      "not '${spring.security.oauth2.resourceserver.jwt.issuer-uri:}'.equals('')")
+  public JwtDecoder jwtDecoder(String issuer) {
```

Comment:
Please make sure to test 
A) If authentication is disabled and issue uri is empty ,  things work - you can do some operation wiht the API
B) If auth is disabled but uris is not empty, then again you can do some operation without authentication 
C) if auth is ENABLED and URI is not empty - you are required authentication token 
D) if uath is ENABLED and URI is empty - some error should show up

Reply:
Tested locally:

A) If authentication is disabled and issue uri is empty , things work.
B) If auth is disabled but uris is not empty, then again you can do some operation without authentication.
C) if auth is ENABLED and URI is not empty - you are required authentication token - tested with tokens from two different issuers, and with non valid token. Authentication works.
D) if auath is ENABLED and URI is empty - some error should show up.